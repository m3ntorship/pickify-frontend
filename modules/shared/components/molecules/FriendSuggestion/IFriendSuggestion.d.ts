declare namespace IFriendSuggestion {
  export interface IProps {
    profilePic: string;
    username: string;
    mutualFriends: number;
  }
}

export { IFriendSuggestion };
