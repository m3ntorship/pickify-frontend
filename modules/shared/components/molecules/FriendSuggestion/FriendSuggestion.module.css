.plus {
  @apply w-10 h-10 rounded-full focus:outline-none bg-primary hover:bg-primary-hover flex justify-center items-center;
}

.delete {
  @apply ml-2 w-10 h-10 rounded-full focus:outline-none border-2 border-grey hover:border-dark-grey flex justify-center items-center;
}
