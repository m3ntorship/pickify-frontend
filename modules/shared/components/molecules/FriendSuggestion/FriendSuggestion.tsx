import React from 'react';
import type { FC, ReactElement } from 'react';
import DeleteIcon from '@modules/shared/components/icons/deleteFriend.svg';
import {
  buttonSizeValues,
  buttonVariantValues,
} from '../../atoms/Button/types/EButton';
import type { IFriendSuggestion } from './IFriendSuggestion';
import PlusIcon from '../../icons/addFriend.svg';
import Button from '../../atoms/Button/Button';
import ImgWithInfoDemo from '../ImgWithInfo copy/ImgWithInfo';

const FriendSuggestion: FC<IFriendSuggestion.IProps> = ({
  profilePic,
  username,
  mutualFriends,
}): ReactElement => {
  const subTitle = `${mutualFriends} ${
    mutualFriends > 1 ? 'Mutual Friends' : 'Mutual Friend'
  }`;

  return (
    <div className="flex justify-between items-center">
      <ImgWithInfoDemo>
        <div className="flex w-min">
          <ImgWithInfoDemo.Image
            avatarSize="medium"
            variant="filled"
            isHidden={false}
            ImageVariant="avatar"
            profilePic={profilePic}
          />
          <ImgWithInfoDemo.Info classes="max-w-13xl flex flex-col justify-between ml-4 whitespace-nowrap">
            <h1 className="text-sm font-normal text-dark">{username}</h1>
            <p className="text-dark-grey text-xs font-light">{subTitle}</p>
          </ImgWithInfoDemo.Info>
        </div>
      </ImgWithInfoDemo>
      <div className="flex">
        <div className="mr-2">
          <Button
            onlyIcon
            size={buttonSizeValues.SMALL}
            variant={buttonVariantValues.PRIMARY}
          >
            <PlusIcon />
          </Button>
        </div>

        <Button
          onlyIcon
          size={buttonSizeValues.SMALL}
          variant={buttonVariantValues.DELETE}
        >
          <DeleteIcon />
        </Button>
      </div>
    </div>
  );
};

export default FriendSuggestion;
