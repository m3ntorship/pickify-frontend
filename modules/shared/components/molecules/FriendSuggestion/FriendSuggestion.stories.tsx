import type { ReactElement } from 'react';
import type { Story } from '@storybook/react';
import FriendSuggestion from './FriendSuggestion';
import type { IFriendSuggestion } from './IFriendSuggestion';

export default {
  component: FriendSuggestion,
  title: 'Molecules/FriendSuggestion',
  argTypes: {
    profile_pic: {
      control: {
        type: 'text',
      },
    },
    username: {
      control: {
        type: 'text',
      },
    },
    mutualFriends: {
      control: 'number',
    },
  },
};

const Template: Story<IFriendSuggestion.IProps> = (args): ReactElement => (
  <FriendSuggestion {...args} />
);

export const Default = Template.bind({});
Default.args = {
  profilePic: 'https://placeimg.com/640/480/any',
  username: 'Miles, Esther',
  mutualFriends: 2,
};
