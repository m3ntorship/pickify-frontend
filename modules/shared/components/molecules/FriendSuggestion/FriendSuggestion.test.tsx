import * as React from 'react';
import * as renderer from 'react-test-renderer';
import FriendSuggestion from './FriendSuggestion';

describe('Friend Suggestion Component', () => {
  it('Should render correctly', () => {
    const tree = renderer
      .create(
        <FriendSuggestion
          profilePic="https://s3-alpha-sig.figma.com/img/443e/6ac0/0864920f73a1963a9015537cae3da88e?Expires=1629676800&Signature=bQGAkd8egMt437a3uVrvuWZhbojvQXv1gT356NyqCL8OGFlYDlZp-~UtHMvqUq8thdWR~v5wsIuxy8l8l8OZIrGZ6tC1U~Fo90xUy529JxeFQuHGzR26XK4fK8ut~MLo6oVyUMYh7cQH0eU4el8nTBt4mWPLigGbpIIt95qVD9wnUaIpG7qyNkBKLtaQXDLuI4pGkQ9urDZBaq9tYVt~uD576PPH~R4v4NF~XHXyeBKNo14KC3DO5zATf~AZystGgqxSZ-XZbru8eJgPaOswzaVy8kk-JHlTzpqU4ABd-M8hC1leUK8OahLfFdDqipZzw00faOJ4iof8pIxjD4PoEg__&Key-Pair-Id=APKAINTVSUGEWH5XD5UA"
          username="Ahmed Adel"
          mutualFriends={5}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
