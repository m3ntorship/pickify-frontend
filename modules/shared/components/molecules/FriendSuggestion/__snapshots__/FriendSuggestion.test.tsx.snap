// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Friend Suggestion Component Should render correctly 1`] = `
<div
  className="flex justify-between items-center"
>
  <div
    className="box"
  >
    <div
      className="flex w-min"
    >
      <div
        className="header-wrapper"
      >
        <div
          className="header"
        >
          <div
            className="wrapper medium"
          >
            <div
              style={
                Object {
                  "boxSizing": "border-box",
                  "display": "inline-block",
                  "margin": 0,
                  "maxWidth": "100%",
                  "overflow": "hidden",
                  "position": "relative",
                }
              }
            >
              <div
                style={
                  Object {
                    "boxSizing": "border-box",
                    "display": "block",
                    "maxWidth": "100%",
                  }
                }
              >
                <img
                  alt=""
                  aria-hidden={true}
                  role="presentation"
                  src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDAiIGhlaWdodD0iNDAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                  style={
                    Object {
                      "border": "none",
                      "display": "block",
                      "margin": 0,
                      "maxWidth": "100%",
                      "padding": 0,
                    }
                  }
                />
              </div>
              <img
                className="absolute w-full h-full"
                decoding="async"
                onClick={[Function]}
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "visibility": "hidden",
                    "width": 0,
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="max-w-13xl flex flex-col justify-between ml-4 whitespace-nowrap"
      >
        <h1
          className="text-sm font-normal text-dark"
        >
          Ahmed Adel
        </h1>
        <p
          className="text-dark-grey text-xs font-light"
        >
          5 Mutual Friends
        </p>
      </div>
    </div>
  </div>
  <div
    className="flex"
  >
    <div
      className="mr-2"
    >
      <button
        className="btn btn-primary btn-small btn-rounded"
        disabled={false}
        type="button"
      >
        <span>
          <svg
            data-file-name="SvgAddFriend"
          />
        </span>
      </button>
    </div>
    <button
      className="btn btn-delete btn-small btn-rounded"
      disabled={false}
      type="button"
    >
      <span>
        <svg
          data-file-name="SvgDeleteFriend"
        />
      </span>
    </button>
  </div>
</div>
`;
